plugin/general/editor.go

```diff
@@ -71,8 +71,9 @@ func EditComposeFile(ctx *context.Context, file string, executorId string, taskI
 	return file, ports, err
 }
 
-func UpdateServiceSessions(serviceName, file, executorId, taskId string, filesMap *types.ServiceDetail, ports *list.Element) (*list.Element, error) {
-	containerDetails, ok := (*filesMap)[file][types.SERVICES].(map[interface{}]interface{})[serviceName].(map[interface{}]interface{})
+func updateServiceSessions(serviceName, file, executorId, taskId string, filesMap types.ServiceDetail, ports *list.Element,
+	extraHosts map[interface{}]bool) (*list.Element, error) {
+	containerDetails, ok := filesMap[file][types.SERVICES].(map[interface{}]interface{})[serviceName].(map[interface{}]interface{})
```

Comment:
Out of curiosity (also this is existing code), what if the first cast fails? That would be a runtime panic, no? Is that not a possibility?

Reply:
this shouldn't cause any runtime panic, we are checking the first cast before invoking function updateServiceSessions.